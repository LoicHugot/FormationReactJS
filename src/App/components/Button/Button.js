import React from 'react';
import PropTypes from 'prop-types';
import './Button.css'

const Button=(props) => {
    console.log(props);
    return <div style={{...props.style,backgroundColor:props.couleurDeFond,fontSize:props.taillePolice+'px'}} className="Button" onClick={(evt)=>{props.onClick('Cliked')}}>{props.label}</div>;
};
Button.propTypes={
    label:PropTypes.string.isRequired,
    couleurDeFond:PropTypes.string,
    taillePolice:PropTypes.number,
    onClick:PropTypes.func.isRequired,    
}
Button.defaultProps={
    taillePolice:32,
    couleurDeFond:'darkGreen'
}
export default Button;