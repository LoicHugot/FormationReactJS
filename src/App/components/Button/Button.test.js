import { render, screen } from '@testing-library/react';
import { Button } from './Button'

describe('<Button />', () => {
  
  test('is button mount', () => {
    render(<Button value="But" onClick={()=>{}}/>);
    const screenButton=screen.getByTestId('Button');
    expect(screenButton).toBeInTheDocument();
    })

    test('is button not empty', () => {
        render(<Button value="But" onClick={()=>{}}/>);
        const screenButton=screen.getByTestId('Button');
        expect(screenButton).not.toBeEmptyDOMElement();
    })
})
