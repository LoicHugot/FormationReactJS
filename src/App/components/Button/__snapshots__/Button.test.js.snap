// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> is button mount 1`] = `
<UNDEFINED
  onClick={[Function]}
  value="But"
/>
`;
